packages/block-library/src/heading/transforms.js

```diff
@@ -12,7 +12,6 @@ const transforms = {
 	from: [
 		{
 			type: 'block',
-			isMultiBlock: true,
```

Comment:
Even in the normal transform menu, transforming multiple paragraphs to multiple headings doesn't really make sense.